### PR TITLE
development.md: rewrite for clarity, grammar, and spelling

### PR DESCRIPTION
**Pull request by alexandercampbell**
Rewrite the `development.md` document to improve clarity and delete redundant information. Along the way, resolve grammar and spelling issues. Wrap to a semi-uniform line width.

### CODE REVIEW

**Comment by alexandercampbell** [comment]
Would you mind taking a look? :)

File: contributors/devel/development.md [hunk 1]
```diff
@@ -119,36 +123,30 @@ Please [install it locally][etcd-install] to run local integration tests.
...
-| | [1.8][go-1.8] not verified as of Feb 2017 |
 
-After installation, you'll need `GOPATH` defined,
-and `PATH` modified to access your Go binaries.
+| Kubernetes     | requires Go                               |
+|----------------|-------------------------------------------|
+| 1.0 - 1.2      | 1.4.2                                     |
+| 1.3, 1.4       | 1.6                                       |
+| 1.5 and higher | 1.7 - 1.7.5                               |
+|                | [1.8][go-1.8] not verified as of Feb 2017 |
```
Comment (pwittrock) [inline contributors/devel/development.md]: 1.7 requires go 1.81 and higher

**Comment by pwittrock** [comment]
LGTM, one comment on the go version

File: contributors/devel/development.md (hunk 1 above)
Comment (alexandercampbell) [inline contributors/devel/development.md]: Updated. Thanks!

**Pull request merged**